### PR TITLE
Adds docker compose for configured ilias

### PR DESCRIPTION
**Pull request by mavogel**
Awesome repo Kevin :) Helped me a lot for my client lib. 

This PR add the following:
- adds `docker-compose` outline for the `ilias-configured` version with a MySQL running in another container
- stores the data of the database in the folder `./.data/db`
- ensures the external db is up before the ilias-server initialized the db with the `wait.sh` script
- updates README

### CODE REVIEW

File: dockerfiles/ilias-configured/DockerfileExtDB [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+RUN chmod -R 775 /var/www/html/ilias/data
+RUN chmod 666 /var/www/html/ilias/ilias.ini.php
+RUN chmod 666 /var/www/html/ilias/data/myilias/client.ini.php
+
+RUN mkdir -p /opt/iliasdata/myilias/files
+RUN mkdir -p /opt/iliasdata/myilias/forum
+RUN mkdir -p /opt/iliasdata/myilias/lm_data
+RUN mkdir -p /opt/iliasdata/myilias/mail
+RUN chown -R www-data:www-data /opt/iliasdata
+RUN chmod -R 751 /opt/iliasdata
```
Comment (kekru) [inline dockerfiles/ilias-configured/DockerfileExtDB]: We should avoid to call RUN more often than needed, as every RUN creates a new layer in the Docker image. 
Better like the following:
```bash
RUN mkdir -p /opt/iliasdata/myilias/files \
 && mkdir -p /opt/iliasdata/myilias/forum \
 && mkdir -p /opt/iliasdata/myilias/lm_data \
...
```

File: dockerfiles/ilias-configured/DockerfileExtDB [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+RUN chown www-data:www-data /data/resources/configured/entrypoint.sh
+RUN chmod 751 /data/resources/configured/entrypoint.sh
+
+COPY ./wait.sh /usr/local/bin/
+RUN chmod +x /usr/local/bin/wait.sh
+ENTRYPOINT ["/usr/local/bin/wait.sh"]
```
Comment (kekru) [inline dockerfiles/ilias-configured/DockerfileExtDB]:
```bash
COPY ./wait.sh /usr/local/bin/
RUN chmod +x /usr/local/bin/wait.sh
ENTRYPOINT ["/usr/local/bin/wait.sh"]
```
These lines are the only new lines in comparison to the old Dockerfile right?
We could also add something like that to the `/data/resources/configured/entrypoint.sh`
```bash
if [ ! -z "$WAIT_FOR" ]; then
     wait.sh
fi
...
```

Then we would only have one Dockerfile which calls the wait.sh-script if the environmental variable is set.

File: dockerfiles/ilias-configured/wait.sh [hunk 3]
```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+should_run=true
+
+after_boot() {
```
Comment (kekru) [inline dockerfiles/ilias-configured/wait.sh]: what is after_boot and before_shutdown used for?

File: dockerfiles/ilias-configured/wait.sh [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        chmod +x $BEFORE_SHUTDOWN
+        "$BEFORE_SHUTDOWN"
+        echo "Before shutdown exited with code $?"
+        exit $?
+    fi
+}
+
+wait_for() {
+    if [ ! -z "$WAIT_FOR" ]; then
+        for i in $(echo $WAIT_FOR | tr "," "\n")
```
Comment (kekru) [inline dockerfiles/ilias-configured/wait.sh]: a description for the syntax of the WAIT_FOR variable would be great in the README.md

File: dockerfiles/ilias-configured/wait.sh [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        do
+          array=$(echo $i | tr ":" "\n")
+          host="${array[0]}"
+          port="${array[1]}"
+          code=1
+          while [ "$code" != "0" ] && [ should_run ]
```
Comment (kekru) [inline dockerfiles/ilias-configured/wait.sh]: Maybe you can shorten this to 
```bash
while ! $(nc -z -v -w5 $host $port)
do
    echo "Waiting for $i to come up..."
    sleep 5
done
```

For another project it worked for me like this: https://github.com/kekru/docker-wait-tcp/blob/master/wait

File: dockerfiles/ilias-configured/wait.sh [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        done
+    fi;
+}
+
+trap before_shutdown HUP INT TERM KILL
+
+wait_for;
+
+echo "Starting command $@..."
+"$@" &
```
Comment (kekru) [inline dockerfiles/ilias-configured/wait.sh]: Can be removed if we put the wait.sh inside the entrypoint, as described in my comment above

**Review by kekru — COMMENTED** [review]
Hey @mavogel thanks for your PR. It's a good point to add a docker-compose file.
Here are a few thoughts

File: dockerfiles/ilias-configured/DockerfileExtDB (hunk 1 above)
Comment (mavogel) [reply dockerfiles/ilias-configured/DockerfileExtDB]: Yes true. Will fix that.

File: dockerfiles/ilias-configured/DockerfileExtDB (hunk 2 above)
Comment (mavogel) [reply dockerfiles/ilias-configured/DockerfileExtDB]: I need to try if that works...

File: dockerfiles/ilias-configured/wait.sh (hunk 3 above)
Comment (mavogel) [reply dockerfiles/ilias-configured/wait.sh]: Well, it's from an internal script we use here at the company to ensure the container are up until the depending one starts. But yes, good question. I'll will try to simplify that and document it.

File: dockerfiles/ilias-configured/wait.sh (hunk 5 above)
Comment (mavogel) [reply dockerfiles/ilias-configured/wait.sh]: Yes, we could also use your wait script to achieve the desired effect of depending containers. I don't mind :)

Comment (kekru) [reply dockerfiles/ilias-configured/wait.sh]: I don't know which one works better, it just worked the last time :)

**Comment by mavogel** [comment]
@kekru I fixed all your concerns :) thx for the feedback

**Comment by kekru** [comment]
@mavogel Looks very good :)
Can we also delete DockerfileExtDB and add the new lines to the old Dockerfile?
```
COPY ./wait.sh /usr/local/bin/
RUN chmod +x /usr/local/bin/wait.sh
```
Then we only have one Dockerfile that we have to keep up to date in the future.

Can you move the `wait.sh` to the resources directory `https://github.com/kekru/ilias-docker/tree/master/dockerfiles/ilias-configured/resources`? That looks more clean, I think :)

**Comment by mavogel** [comment]
done :)

**Comment by kekru** [comment]
Looks very good, thank you :)
